app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,18 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time, threshold=1.day)
+    raise _("Start time must be earlier than End time") if start_time > end_time
+    # Create an array of ordered pairs from start_time and end_time so that each ordered pair is contained
+    # within the threshold.  Then, reverse it so the newest ordered pair is first:
+    # start_time = 2017/01/01 12:00:00, end_time = 2017/01/04 12:00:00
+    # [[2017-01-03 12:00:00 UTC, 2017-01-04 12:00:00 UTC],
+    #  [2017-01-02 12:00:00 UTC, 2017-01-03 12:00:00 UTC],
+    #  [2017-01-01 12:00:00 UTC, 2017-01-02 12:00:00 UTC]]
+    (start_time..end_time).step_value(threshold).each_cons(2).collect { |s_time, e_time| [s_time, e_time] }.reverse
```

Comment:
One thing I could do is pass in the `interval_name` as an argument and forego the extra call to [`map` below](https://github.com/ManageIQ/manageiq/pull/14332/files/9e06dbef7d5e1f39b49060cbb64cba1115dc909b#diff-98e67a56bb0805013a9953cc93e47389R59).

Then, the `collect` call here really is manipulating the tuples.